File: Semestrul 4/MPP/Proiecte/c#_gui/domain/Voluntar.cs
namespace csharp_avalonia.domain;

public class Voluntar : Entity<int>
{
    private string username;
    private string parola;
    private string nume_voluntar;

    public Voluntar(int id, string username, string password, string nume_voluntar)
    {
        this.id = id;
        this.username = username;
        this.parola = password;
        this.nume_voluntar = nume_voluntar;
    }

    public Voluntar(string username, string password, string nume_voluntar)
    {
        this.username = username;
        this.parola = password;
        this.nume_voluntar = nume_voluntar;
    }
    
    
    public string Username { get => username; set => username = value; }
    
    public string Password { get => parola; set => parola = value; }
    
    public string Nume_voluntar { get => nume_voluntar; set => nume_voluntar = value; }
}

File: Semestrul 4/MPP/Proiecte/jwt - tema 8/Voluntar.java
package model;

import jakarta.persistence.*;
@jakarta.persistence.Entity(name = "Voluntar")
@Table(name = "VOLUNTARI")
public class Voluntar{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_voluntar")
    private Integer id;

    @Column(name = "username", nullable = false, unique = true)
    private String username;

    @Column(name = "parola", nullable = false)
    private String password;

    @Column(name = "nume_voluntar", nullable = false)
    private String nume_voluntar;

    public Voluntar() {}

    public Voluntar(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public Voluntar(int id, String username, String password, String nume_voluntar) {
        this.id = id;
        this.username = username;
        this.password = password;
        this.nume_voluntar = nume_voluntar;
    }

    public Voluntar(String username, String password, String nume_voluntar) {
        this.username = username;
        this.password = password;
        this.nume_voluntar = nume_voluntar;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getNume_voluntar() {
        return nume_voluntar;
    }

    public void setNume_voluntar(String nume_voluntar) {
        this.nume_voluntar = nume_voluntar;
    }

    @Override
    public String toString() {
        return nume_voluntar;
    }

}
